Standardize Translation Method

lawsj opened the pull request:
# Description

Made changes to use the new  "useTranslate()" instead of the old "translate()".

Partly Addresses #1204 

## Type of change

(Check the ones that apply by placing an "x" instead of the space in the [ ] so it becomes [x])

- [ ] Note merging this changes the database configuration.
- [ ] This change requires a documentation update

## Checklist

(Note what you have done by placing an "x" instead of the space in the [ ] so it becomes [x]. It is hoped you do all of them.)

- [x] I have followed the [OED pull request](https://openenergydashboard.github.io/developer/pr.html) ideas
- [x] I have removed text in ( ) from the issue request
- [x] You acknowledge that every person contributing to this work has signed the [OED Contributing License Agreement](https://openenergydashboard.github.io/developer/cla.html) and each author is listed in the Description section.

##Limitations

N/a

Commit:
Standardize Translation Method

Rakesh-Ranga-Buram commented:
Everything looks good to me. Please convert the draft to PR.

Rakesh-Ranga-Buram commented:
@huss , this PR only include changes related to translate() method and not formatMessage() method. Since formatMessage() changes are already been done on #1314

Commit:
Standardize translation method

huss commented on src/client/app/components/BarControlsComponent.tsx:
```diff
@@ -89,6 +89,7 @@ export default function BarControlsComponent() {
 			dispatch(graphSlice.actions.updateBarDuration(moment.duration(value, 'days')));
 		}
 	};
+	const translate = useTranslate();
```
While this comment is here it applies to many files changed. The location of this line varies by file and is not, from my perspective, in the ideal location in the code. It is similar to ``const dispatch = useAppDispatch();`` that is at/near the top of the function. I think that is a good location for this type of code. I think all of these lines should be there (or in the block defining functions as in some files) or near the top of the function if dispatch is not present. If you feel otherwise then please let me know.

Rakesh-Ranga-Buram replied:
@huss , I too see the same type of failure.  After debugging, I see that placing `const translate = useTranslate();` at the top of the function was crashing in some components. However placing it somewhere below was rectifying the error. I was unable to technically debug why this was happening.
On the other hand, the translate changes in some components was resulting in error and that is why some admin pages were crashing. However in this case, altering the location of `const translate = useTranslate();` did not work out.

huss reviewed (changes requested):
Thanks to @lawsj for another contribution. Overall, this works as desired. I made one comment in a file to address. In addition, here are comments that could not be placed in a file:

- src/client/app/utils/translate.ts has the previously used translation function. I could not find it used (which is good). If so, I think this file should be removed to avoid unused code and to stop someone from using the old translate method.
- OED is crashing when I go to a number of admin pages. For example, going to the conversion page and then try to create a conversion does this. What is strange is I cannot see why any change made in this PR would cause this to happen. Thus, I removed the new translate change in this file but it still happened. Next I tried changing to the development branch and it no longer crashed. This seems to imply it was a change in the pull request but why is unclear. Modest debugging did not find the reason. Given this, I wanted to know if you (or anyone else) sees this same type of failure? I want to know if it is a general problem or something in my setup.

huss commented:
This is just for the record: We are continuing to try to determine the cause of the issue. It may well be something outside the direct changes made in this PR.

Rakesh-Ranga-Buram committed:
Standardize Translation Method

Rakesh-Ranga-Buram commented:
@huss , the code has been updated with the discussed changes. Can you please review this PR

huss committed:
Merge remote-tracking branch 'origin/development' into pr/lawsj/1354

huss committed:
add missing csv menu translations

huss committed:
remove map useTranslate

While two functions where in the components directory, they were
not in the function component so the React Hooks version of
translate should not be used. Before the change it was getting errors
and crashing. now it seems fine.

huss commented:
I pushed commits to do the following:

1. Merge in development and address conflicts.
2. Added missing CSV menu translations. Not directly related to this PR but similar so did it.
3. Removed a couple of uses of useTranslate from maps because outside function component and causing issues.

huss reviewed (approved):
Thanks to @Rakesh-Ranga-Buram for the updated code. Review and testing found it works as desired at this time. Congratulations to @lawsj on their first accepted contribution to OED.
